blog/2023-05-11-conda-doctor.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "conda doctor : Detecting conda environment inconsistencies easily"
+slug: "2023-05-11-conda-doctor"
+description: |
+    The new `conda doctor` command allows users to detect packages with missing files in their conda environments.
+
+authors: [forgottenprogramme]
+tags: [conda, environments]
+
+---
+The newest `conda doctor` command will be available with the latest conda 23.5.0, enabling conda users to detect corrupted packages (packages with files missing) in their conda environment quickly.
+
+One way packages in an environment may go corrupt is when one or more files in that package are accidentally deleted. Even a single missing file leads to a corrupted package.
+Until now there was no way to learn of these missing files except manually checking for their existence. With `conda doctor`, users can learn about all such packages in their conda environment with a single short command. 
+
+## How Does It Work?
+Every conda environment has a `conda-meta` directory associated with it, which keeps a log of all the packages (and all the files in them) ever installed into that environment. 
+`conda doctor` parses these metadata files, checks for the existence of each of the listed files and reports about the missing ones. 
+This is extremely helpful as it is otherwise very difficult to know that a package has missing files and to detect these missing files in the package. 
+
+## The Plugin Mechanism 
+`conda doctor` is implemented using conda’s new [plugins mechanism](https://www.anaconda.com/blog/introducing-a-new-plugin-mechanism-for-conda/), utilizing the "subcommand plugin hook" and comes bundled with conda itself.
+
+Often the issues that conda users run into are issues in their conda environment. The idea behind implementing a `conda doctor` command is to "diagnose" the inconsistencies in a conda environment and report to the user, therefore empowering the user to better understand the cause of the issues they are experiencing.
+
+## A Nine Year Old Issue
+What’s interesting is that [this feature was requested by Peter Wang](https://github.com/conda/conda/issues/474), the CEO of Anaconda, nine years ago in the early days of conda. A number of features requested from `conda doctor` back then ended up being implemented elsewhere in conda over the years. 
```

Comment:
```suggestion
What’s interesting is that [this feature was requested by Peter Wang](https://github.com/conda/conda/issues/474), the CEO of Anaconda, nine years ago in the early days of conda. A number of features requested via this `conda doctor` issue back then ended up being implemented elsewhere in conda over the years.
```

Reply:
Not totally sure if this is the case, but if it is, wanted to make it more clear. I assume the features outlined in that linked issue are what you are meaning?

Reply:
Yes.